runtime/src/mem/jemalloc/mem-jemalloc.c

```diff
@@ -28,15 +28,40 @@
 #include "chpl-comm.h"
 #include "chpl-mem.h"
 #include "chplmemtrack.h"
+#include "chplsys.h"
+#include "chpl-topo.h"
 #include "chpltypes.h"
 #include "error.h"
 
-static struct shared_heap {
+
+//
+// Information about shared heaps gotten from the comm layer.
+//
+typedef struct shared_heap {
   void* base;
   size_t size;
   size_t cur_offset;
   pthread_mutex_t alloc_lock;
-} heap; // static, will be "zero" initialized automatically
+} shared_heap_t;
+
+static shared_heap_t* heaps;
+static int num_heaps;
+
+static inline int get_num_heaps(void) {
+  return num_heaps;
```

Comment:
And my motivation here is that I have such a hard time following when a global has been initialized, and when it's safe to use. I much prefer avoiding that unless there's a big performance reason to do so.

Also even if get_num_heaps was called concurrently, I don't think that'd be a problem. You might end up initializing concurrently, but I don't think that'd hurt anything.